### ISSUE TITLE
Bleadperl v5.25.9-10-g3a0fe090c1 breaks SPROUT/JE-0.066.tar.gz

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#15832]
Migrated from [rt.perl.org#130639](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=130639) (status was 'resolved')

Searchable as RT130639$

**Comment by p5pRT** [comment]
### From @andk

bisect


---
commit 3a0fe090c1f24f4a3748d00616b87eb4c8dd4475
Author&#8203;: Zefram \<zefram@&#8203;fysh\.org>
Date&#8203;:   Sun Jan 22 03&#8203;:20&#8203;:08 2017 \+0000

&nbsp;    handle errors in gen\_constant\_list

diagnostics


---
Fail report shortly after commit&#8203;: http&#8203;://www\.cpantesters\.org/cpan/report/2f3bb378\-e177\-11e6\-af27\-bba69f28caea
Pass report immediately before commit&#8203;: http&#8203;://www\.cpantesters\.org/cpan/report/8bca308e\-e077\-11e6\-a267\-6a5f9f28caea

perl \-V


---
Summary of my perl5 \(revision 5 version 25 subversion 10\) configuration&#8203;:
&nbsp;  Commit id&#8203;: 3a0fe090c1f24f4a3748d00616b87eb4c8dd4475
&nbsp;  Platform&#8203;:
&nbsp;    osname=linux
&nbsp;    osvers=4\.8\.0\-2\-amd64
&nbsp;    archname=x86\_64\-linux
&nbsp;    uname='linux k93msid 4\.8\.0\-2\-amd64 \#1 smp debian 4\.8\.11\-1 \(2016\-12\-02\) x86\_64 gnulinux '
&nbsp;    config\_args='\-Dprefix=/home/sand/src/perl/repoperls/installed\-perls/host/k93msid/v5\.25\.9\-10\-g3a0fe090c1/89ad \-Dmyhostname=k93msid \-Dinstallusrbinperl=n \-Uversiononly \-Dusedevel \-des \-Ui\_db \-Dlibswanted=cl pthread socket inet nsl gdbm dbm malloc dl ld sun m crypt sec util c cposix posix ucb BSD gdbm\_compat \-Uuseithreads \-Uuselongdouble \-DDEBUGGING=\-g'
&nbsp;    hint=recommended
&nbsp;    useposix=true
&nbsp;    d\_sigaction=define
&nbsp;    useithreads=undef
&nbsp;    usemultiplicity=undef
&nbsp;    use64bitint=define
&nbsp;    use64bitall=define
&nbsp;    uselongdouble=undef
&nbsp;    usemymalloc=n
&nbsp;    bincompat5005=undef
&nbsp;  Compiler&#8203;:
&nbsp;    cc='cc'
&nbsp;    ccflags ='\-fwrapv \-fno\-strict\-aliasing \-pipe \-fstack\-protector\-strong \-I/usr/local/include \-D\_LARGEFILE\_SOURCE \-D\_FILE\_OFFSET\_BITS=64'
&nbsp;    optimize='\-O2 \-g'
&nbsp;    cppflags='\-fwrapv \-fno\-strict\-aliasing \-pipe \-fstack\-protector\-strong \-I/usr/local/include'
&nbsp;    ccversion=''
&nbsp;    gccversion='6\.2\.1 20161124'
&nbsp;    gccosandvers=''
&nbsp;    intsize=4
&nbsp;    longsize=8
&nbsp;    ptrsize=8
&nbsp;    doublesize=8
&nbsp;    byteorder=12345678
&nbsp;    doublekind=3
&nbsp;    d\_longlong=define
&nbsp;    longlongsize=8
&nbsp;    d\_longdbl=define
&nbsp;    longdblsize=16
&nbsp;    longdblkind=3
&nbsp;    ivtype='long'
&nbsp;    ivsize=8
&nbsp;    nvtype='double'
&nbsp;    nvsize=8
&nbsp;    Off\_t='off\_t'
&nbsp;    lseeksize=8
&nbsp;    alignbytes=8
&nbsp;    prototype=define
&nbsp;  Linker and Libraries&#8203;:
&nbsp;    ld='cc'
&nbsp;    ldflags =' \-fstack\-protector\-strong \-L/usr/local/lib'
&nbsp;    libpth=/usr/local/lib /usr/lib/gcc/x86\_64\-linux\-gnu/6/include\-fixed /usr/include/x86\_64\-linux\-gnu /usr/lib /lib/x86\_64\-linux\-gnu /lib/\.\./lib /usr/lib/x86\_64\-linux\-gnu /usr/lib/\.\./lib /lib
&nbsp;    libs=\-lpthread \-lnsl \-ldl \-lm \-lcrypt \-lutil \-lc
&nbsp;    perllibs=\-lpthread \-lnsl \-ldl \-lm \-lcrypt \-lutil \-lc
&nbsp;    libc=libc\-2\.24\.so
&nbsp;    so=so
&nbsp;    useshrplib=false
&nbsp;    libperl=libperl\.a
&nbsp;    gnulibc\_version='2\.24'
&nbsp;  Dynamic Linking&#8203;:
&nbsp;    dlsrc=dl\_dlopen\.xs
&nbsp;    dlext=so
&nbsp;    d\_dlsymun=undef
&nbsp;    ccdlflags='\-Wl\,\-E'
&nbsp;    cccdlflags='\-fPIC'
&nbsp;    lddlflags='\-shared \-O2 \-g \-L/usr/local/lib \-fstack\-protector\-strong'


Characteristics of this binary \(from libperl\)&#8203;: 
&nbsp;  Compile\-time options&#8203;:
&nbsp;    HAS\_TIMES
&nbsp;    PERLIO\_LAYERS
&nbsp;    PERL\_COPY\_ON\_WRITE
&nbsp;    PERL\_DONT\_CREATE\_GVSV
&nbsp;    PERL\_MALLOC\_WRAP
&nbsp;    PERL\_OP\_PARENT
&nbsp;    PERL\_PRESERVE\_IVUV
&nbsp;    PERL\_USE\_DEVEL
&nbsp;    USE\_64\_BIT\_ALL
&nbsp;    USE\_64\_BIT\_INT
&nbsp;    USE\_LARGE\_FILES
&nbsp;    USE\_LOCALE
&nbsp;    USE\_LOCALE\_COLLATE
&nbsp;    USE\_LOCALE\_CTYPE
&nbsp;    USE\_LOCALE\_NUMERIC
&nbsp;    USE\_LOCALE\_TIME
&nbsp;    USE\_PERLIO
&nbsp;    USE\_PERL\_ATOF
&nbsp;  Built under linux
&nbsp;  Compiled at Jan 22 2017 03&#8203;:43&#8203;:33
&nbsp;  @&#8203;INC&#8203;:
&nbsp;    /home/sand/src/perl/repoperls/installed\-perls/host/k93msid/v5\.25\.9\-10\-g3a0fe090c1/89ad/lib/site\_perl/5\.25\.10/x86\_64\-linux
&nbsp;    /home/sand/src/perl/repoperls/installed\-perls/host/k93msid/v5\.25\.9\-10\-g3a0fe090c1/89ad/lib/site\_perl/5\.25\.10
&nbsp;    /home/sand/src/perl/repoperls/installed\-perls/host/k93msid/v5\.25\.9\-10\-g3a0fe090c1/89ad/lib/5\.25\.10/x86\_64\-linux
&nbsp;    /home/sand/src/perl/repoperls/installed\-perls/host/k93msid/v5\.25\.9\-10\-g3a0fe090c1/89ad/lib/5\.25\.10
&nbsp;    \.


\-\- 
andreas

**Comment by p5pRT** [comment]
### From zefram@fysh.org

>Subject&#8203;: \[perl \#130639\] Bleadperl v5\.25\.9\-10\-g3a0fe090c1 breaks
> SPROUT/JE\-0\.066\.tar\.gz

In my own tests\, JE\-0\.066 fails both before and after the commit\, in
identical fashion\.  There's a lot of

&nbsp;    perl&#8203;: utf8\.c&#8203;:1418&#8203;: Perl\_utf8n\_to\_uvchr\_error&#8203;: Assertion \`0' failed\.

So I'm dubious about the identification of the breaking commit\.

\-zefram

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @khwilliamson

On Wed\, 25 Jan 2017 13&#8203;:31&#8203;:26 \-0800\, zefram@&#8203;fysh\.org wrote&#8203;:
> >Subject&#8203;: \[perl \#130639\] Bleadperl v5\.25\.9\-10\-g3a0fe090c1 breaks
> > SPROUT/JE\-0\.066\.tar\.gz
> 
> In my own tests\, JE\-0\.066 fails both before and after the commit\, in
> identical fashion\.  There's a lot of
> 
>     perl&#8203;: utf8\.c&#8203;:1418&#8203;: Perl\_utf8n\_to\_uvchr\_error&#8203;: Assertion \`0' failed\.
> 
> So I'm dubious about the identification of the breaking commit\.
> 
> \-zefram
> 

Those asserts are to catch in DEBUGGING builds violations of the changed API\.  It is now illegal to call Perl\_utf8n\_to\_uvchr with input length 0\.  If you have nothing to decode\, don't try to decode\.
\-\- 
Karl Williamson

**Comment by p5pRT** [comment]
### From zefram@fysh.org

Suppressing the utf8 assertions, I think I've got to what this ticket
is about.  The commit identification is fine.  The problem reduces to:

```
$ good_perl -lwe 'print ":".."~"; print 1'
:
1
$ bad_perl -lwe 'print ":".."~"; print 1'
Argument ":" isn't numeric in range (or flop) at -e line 1.
$
```

-zefram

**Comment by p5pRT** [comment]
### From zefram@fysh.org

Fixed by b3698342565fb462291fba4b432cfcd05b6eb4e1\.

\-zefram

**Comment by p5pRT** [comment]
### From @iabyn

On Fri\, Jan 27\, 2017 at 04&#8203;:44&#8203;:02AM \+0000\, Zefram wrote&#8203;:
> Fixed by b3698342565fb462291fba4b432cfcd05b6eb4e1\.

\.\.\. so I'm closing the ticket\.

\-\- 
"Strange women lying in ponds distributing swords is no basis for a system
of government\. Supreme executive power derives from a mandate from the
masses\, not from some farcical aquatic ceremony\."
&nbsp;    \-\- Dennis\, "Monty Python and the Holy Grail"

**Comment by p5pRT** [comment]
@iabyn - Status changed from 'open' to 'resolved'
